diffusion/inference.py

```diff
@@ -0,0 +1,80 @@
+import argparse
+import os
+import torch
+
+from model import DiffusionModel
+from unet import Unet
+from utils import save_samples
+
+
+parser = argparse.ArgumentParser(description='Diffusion Model Inference')
+parser.add_argument('--ckpt', required=True, type=str, help="Pretrained checkpoint")
+parser.add_argument('--num-images', default=8, type=int, help="Number of images per iteration")
+parser.add_argument('--sampling-method', default="ddpm", type=str, help="Sampling method (ddpm vs ddim")
+args = parser.parse_args()
+
+
+@torch.no_grad()
+def ddim_sample(diffusion_model, num_images=8, image_size=32):
+    """
+    Use the reverse sampling process from DDIM to go from 
+    a tensor of noise to a generated image.
+
+    TODO (Q3.2.1): fill `ddim_sample`
+    """
+    num_channels = diffusion_model.channels
+    return diffusion_model.ddim_sample((num_images, num_channels, image_size, image_size))
+
+
+@torch.no_grad()
+def ddpm_sample(diffusion_model, num_images=8, image_size=32):
+    """
+    Use the reverse sampling process from DDIM to go from 
+    a tensor of noise to a generated image.
+
+    TODO (Q3.1): fill `ddpm_sample` and the related functions.
+    """
+    num_channels = diffusion_model.channels
+    return diffusion_model.ddpm_sample((num_images, num_channels, image_size, image_size))
+
+def main(
```

Comment:
all of this can be moved to __main__ block instead of having a main function. let's not specify default paths for the checkpoint either since it won't be part of the repo.